saleor/product/models.py

```diff
@@ -368,8 +369,7 @@ def plain_text_description(self) -> str:
         return json_content_to_raw_text(self.description)
 
     def get_first_image(self):
-        images = list(self.images.all())
-        return images[0] if images else None
+        return self.media.filter(type=ProductMediaTypes.IMAGE).first() or None
```

Comment:
@GrzegorzDerdak By calling .filter here, you cancel all prefetched that we did. You should use .all() and iterate over the list to find the first image, without this, the logic that uses prefetched `images` will get a lot of additional DB queries.
![image](https://user-images.githubusercontent.com/23282005/107200625-39f38a80-69f8-11eb-929e-454bd092eec5.png)